File: src/modifiers/textCase.ts
import { createModifier } from './createModifier';

/// SwiftUI `Text.Case` values. `null` clears any inherited case
/// transform (SwiftUI's `nil`).
export type TextCase = 'uppercase' | 'lowercase';

export interface TextCaseParams {
  value?: TextCase | null;
}

/// SwiftUI `.textCase(_:)`. Applies a case transform to text within the
/// view. Pass `'uppercase'`/`'lowercase'`, `null` to clear it, or an
/// object.
export function textCase(params?: TextCaseParams | TextCase | null) {
  if (typeof params === 'string' || params === null) {
    return createModifier('textCase', { value: params });
  }
  return createModifier('textCase', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/TextCase.swift
import SwiftUI

/// SwiftUI `.textCase(_:)`. Applies a case transform to text within the
/// view. `'uppercase'`/`'lowercase'` map to the matching `Text.Case`;
/// anything else (including an explicit `null`) passes `nil`, which clears
/// any inherited transform.
enum RNWTextCaseModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("textCase") { view, params, _ in
            let textCase: Text.Case?
            switch params.string("value") {
            case "uppercase": textCase = .uppercase
            case "lowercase": textCase = .lowercase
            default:          textCase = nil
            }
            return AnyView(view.textCase(textCase))
        }
    }
}
